File: QuranMac/Sources/QuranMac/Models/Verse.swift
import Foundation

struct Verse: Identifiable, Codable, Hashable {
    let id: Int
    let verseNumber: Int
    let verseKey: String
    let textUthmani: String
    
    enum CodingKeys: String, CodingKey {
        case id
        case verseNumber = "verse_number"
        case verseKey = "verse_key"
        case textUthmani = "text_uthmani"
    }
}

struct VersesResponse: Codable {
    let verses: [Verse]
    let pagination: Pagination
}

struct Pagination: Codable {
    let totalRecords: Int
    
    enum CodingKeys: String, CodingKey {
        case totalRecords = "total_records"
    }
}

File: QuranWindows/QuranWindows/Models/Verse.cs
using Newtonsoft.Json;

namespace QuranWindows.Models;

public class Verse
{
    [JsonProperty("id")]
    public int Id { get; set; }

    [JsonProperty("verse_number")]
    public int VerseNumber { get; set; }

    [JsonProperty("verse_key")]
    public string VerseKey { get; set; } = string.Empty;

    [JsonProperty("text_uthmani")]
    public string TextUthmani { get; set; } = string.Empty;
}
